xbmc/cores/AudioEngine/Sinks/AESinkAUDIOTRACK.cpp

```diff
@@ -533,16 +532,11 @@ bool CAESinkAUDIOTRACK::Initialize(AEAudioFormat &format, std::string &device)
       double period_time =
           static_cast<double>(period_size) / (m_sink_frameSize * m_sink_sampleRate);
 
-      while (period_time > 0.064)
+      while (period_time > max_time)
       {
         period_time /= 2;
         period_size /= 2;
       }
-      while (period_time < 0.032)
```

Comment:
I'm not sure of remove this: is not still useful to guarantee minimum period_time of 32ms?

ie if buffer is now 128 ms  guarantee 4 x 32ms = 128 ms  

and not   8 x 16 ms ???

Reply:
The reason is a logical one:
The first loop divides only if the value is above 64 ms, so after the above division no value smaller than 32 ms can result -> therefore the second loop is not needed.

Reply:
This seems introduced here https://github.com/xbmc/xbmc/commit/d9eeac9847409d7c554ec22f872ac129227bc0cc as workaround for MiBoxes. So I understand that it is not necessary in this case either.

Reply:
I see it as a "safety measure" at zero cost

Reply:
Okay - sure. No problem, we can add that. Though we should think about that anyways, cause for DTS-HD-MA 10ms or even for TrueHD we only get 20 ms frames .... so the lower bound should be even lower in practice. I will readd it and change the minimal value to be 16, which in practice we also won't hit ... cause of the first loop.

Reply:
Btw. the real old workaround was this:
https://github.com/xbmc/xbmc/commit/d9eeac9847409d7c554ec22f872ac129227bc0cc#diff-9884fdeb14fb2a0179b4983db00f524e1298005123d633a3b2ed9a261248e1e9R503 <- the else case is not relevant.

Reply:
I will reintroduce this - after we are done with this one ... :-)